elytron/WFLY-17143-request-uri-parameters.adoc

```diff
@@ -0,0 +1,161 @@
+= Add the ability to specify that the OIDC Authentication Request should include request and request_uri parameters
+:author:            Prarthona Paul
+:email:             prpaul@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+OpenID Connect is an authentication mechanism that builds on OAuth 2.0 and allows a user to login to a web application using credentials established by an OpenID provider. Authentication requests sent using OpenID Connect can be signed and optionally encrypted by encoding query parameters in a JWT. This can be achieved using one of two auth request parameters: `request` and `request_uri`, which are both optional. The current version of `elytron-oidc-client` does not allow the user to specify `request` or `request_uri`. Instead, all requests are sent in using the OAuth 2.0 request syntax. This RFE will add an attribute to the `elytron-oidc-client` subsystem called `request-object-required`, which can allow the user to specify if they will use either of the request parameters. 
+
+According to the https://openid.net/specs/openid-connect-core-1_0.html#JWTRequests[OpenID docs], requests using these parameters are represented as JWTs, which are respectively passed by value or by reference. Depending on the OpenID Provider, support for `request` may or may not be available.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-17143[WFLY-17143]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1974[EAP7-1974]
+* https://issues.redhat.com/browse/ELY-2584[ELY-2584]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+* [x] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+* WildFly 
+* WildFly Elytron
+
+=== Other Interested Projects
+
+N/A
+
+=== Relevant Installation Types
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* A new attribute named `request-object-required` will be added to the `secure-deployment` resource under the `elytron-oidc-client` subsystem. The value for this attribute will be a String with 3 acceptable values: 
```

Comment:
> A new attribute named `request-object-required` will be added to the `secure-deployment` resource under the `elytron-oidc-client` subsystem.

Why just to `secure-deployment`? Why not also to `provider`, `realm`, and/or `secure-server`?

Reply:
Hi @OndrejKotek, 
Thank you for your review. Adding this attribute to `secure-server` makes sense and I can add that to the analysis doc. However, I am not sure if this attribute relates to `realm` or `provider_url` since request parameters encrypt query parameters. Additionally, you cannot configure `client-id` using `realm` or `provider_url`, which is needed for the auth request. So, they would have to configure `secure-server` or `secure-deployment` regardless. However, if you see other benefits to having `request-object-required` added to `realm` and `provider_url`, I am happy to add it there!

Reply:
IIUIC, `provider` and `realm` resources are just containers for settings that can be referenced in `secure-deployment` or `secure-server` resources. It makes sense to me to allow to define `request-object-required`also in such resources (at higher/more global configuration levels) like the other attributes. WDYT?